### PR TITLE
Fix some typos in `docs/DafnyRef/README.md`

### PR DESCRIPTION
**Pull request by kazarmy**
### Description

This PR fixes a typo and a couple of repetitions in `docs/DafnyRef/README.md`. The changes were done using the GitHub pencil, which probably added a newline at the end.

### How has this been tested?

Eyeballing the preview.

----

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: docs/DafnyRef/README.md [hunk 1]
```diff
@@ -65,11 +65,9 @@ definition file.
 Github uses rouge, via Jekyll. The syntax definition is a ruby-based file
 maintained in the rouge github repo.
 To modify the definition, follow the directions in
-[https://rouge-ruby.github.io/docs/file.LexerDevelopment.html]
-(https://rouge-ruby.github.io/docs/file.LexerDevelopment.html)
+https://rouge-ruby.github.io/docs/file.LexerDevelopment.html
 after setting up a development environment according to
-[https://rouge-ruby.github.io/docs/file.DevEnvironment.html]
-(https://rouge-ruby.github.io/docs/file.DevEnvironment.html).
+https://rouge-ruby.github.io/docs/file.DevEnvironment.html.
```
Comment (fabiomadge) [inline docs/DafnyRef/README.md:70]: I think the issue might have been the newline, but think we can improve on what was originally intended.
```
To modify the definition, follow [these](https://rouge-ruby.github.io/docs/file.LexerDevelopment.html) directions after [setting up](https://rouge-ruby.github.io/docs/file.DevEnvironment.html) a development environment.
```

Comment (kazarmy) [reply docs/DafnyRef/README.md]: The newline (or lack of it) is an issue? Ok made changes as suggested above 884095a988e72105aefeb236d4e6343d7fd59e0c.

**Review by fabiomadge — APPROVED** [review]
Thanks for bothering!

File: docs/DafnyRef/README.md (hunk 1 above)
Comment (fabiomadge) [reply docs/DafnyRef/README.md]: That’s just my best guess as to why the Markdown processor didn't pick up the link.

Comment (kazarmy) [reply docs/DafnyRef/README.md]: Oh it was those newlines that you were referring to ... yeah I think there cannot be any whitespace between the 2 components of the link i.e. between `[...]` and `(...)`.

**Pull request merged**